server/src/main/java/io/pravega/schemaregistry/storage/client/WireCommandClient.java

```diff
@@ -0,0 +1,417 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.storage.client;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.netty.buffer.Unpooled;
+import io.pravega.auth.AuthenticationException;
+import io.pravega.client.connection.impl.ConnectionPool;
+import io.pravega.client.connection.impl.RawClient;
+import io.pravega.client.control.impl.ModelHelper;
+import io.pravega.client.stream.impl.ConnectionClosedException;
+import io.pravega.client.tables.IteratorItem;
+import io.pravega.client.tables.IteratorState;
+import io.pravega.client.tables.impl.IteratorStateImpl;
+import io.pravega.client.tables.impl.TableSegmentEntry;
+import io.pravega.client.tables.impl.TableSegmentKey;
+import io.pravega.client.tables.impl.TableSegmentKeyVersion;
+import io.pravega.common.Exceptions;
+import io.pravega.common.tracing.TagLogger;
+import io.pravega.schemaregistry.storage.StoreExceptions;
+import io.pravega.shared.protocol.netty.ConnectionFailedException;
+import io.pravega.shared.protocol.netty.Reply;
+import io.pravega.shared.protocol.netty.Request;
+import io.pravega.shared.protocol.netty.WireCommand;
+import io.pravega.shared.protocol.netty.WireCommandType;
+import io.pravega.shared.protocol.netty.WireCommands;
+import lombok.AccessLevel;
+import lombok.Getter;
+import org.slf4j.LoggerFactory;
+
+import java.util.AbstractMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.stream.Collectors;
+
+import static io.pravega.controller.stream.api.grpc.v1.Controller.NodeUri;
+
+/**
+ * Used for making wire command calls into Segment Store.  
+ */
+public class WireCommandClient {
```

Comment:
nit: Would it be useful to create an interface for this, so that it can be easily stubbed/mocked for testing?

Reply:
We have created a mock for this class for testing purposes called WireCommandClientMock.java